src/test/java/seedu/sudowudo/logic/LogicManagerTest.java

```diff
@@ -722,6 +723,50 @@ public void execute_editIndexNotFound_errorMessageShown() throws Exception {
     }
     //@@author
 
+    //@@author A0131560U
+    @Test
+    public void execute_editInvalidDescription_errorMessageShown() throws Exception {
+        TestDataHelper helper = new TestDataHelper();
+        Item initial = helper.workingItemWithDates("valid", LocalDateTime.of(2016, 10, 10, 10, 10, 10),
+                LocalDateTime.of(2016, 11, 11, 11, 11, 11));
+        List<Item> expectedList = helper.generateItemList(initial);
+        helper.addToModel(model, expectedList);
+        TaskBook expectedTB = helper.generateTaskBook(expectedList);
+        String newDescription = "invalid!!";
+        assertCommandBehavior("edit 1 desc:" + newDescription, Description.MESSAGE_NAME_CONSTRAINTS, expectedTB, expectedList);
+    }
+    
+    //@@author A0131560U
+    @Test
+    public void execute_editInvalidDates_errorMessageShown() throws Exception {
+        TestDataHelper helper = new TestDataHelper();
+        Item initial = helper.workingItemWithDates("valid", LocalDateTime.of(2016, 10, 10, 10, 10, 10),
+                                                            LocalDateTime.of(2016, 11, 11, 11, 11, 11));
+        List<Item> expectedList = helper.generateItemList(initial);
+        helper.addToModel(model, expectedList);
+        TaskBook expectedTB = helper.generateTaskBook(expectedList);
+        assertCommandBehavior("edit 1 period: today till yesterday", Item.MESSAGE_DATE_CONSTRAINTS, expectedTB, expectedList);
+        assertCommandBehavior("edit 1 start: january 1st 2017", Item.MESSAGE_DATE_CONSTRAINTS, expectedTB, expectedList);
+        assertCommandBehavior("edit 1 end: january 1st 2015", Item.MESSAGE_DATE_CONSTRAINTS, expectedTB, expectedList);
+    }
+    
+    //@@author A0131560U
+    @Test
+    public void execute_editPeriod_success() throws Exception {
+        TestDataHelper helper = new TestDataHelper();
+        Item initial = helper.workingItemWithDates("valid", LocalDateTime.of(2016, 10, 10, 10, 10, 10),
+                                                            LocalDateTime.of(2016, 11, 11, 11, 11, 11));
+        Item target = helper.workingItemWithDates("valid", LocalDateTime.of(2015, 10, 10, 10, 10, 10),
+                                                            LocalDateTime.of(2015, 11, 11, 11, 11, 11));
+        List<Item> initialList = helper.generateItemList(initial);
+        List<Item> targetList = helper.generateItemList(target);
+        helper.addToModel(model, initialList);
+        TaskBook expectedTB = helper.generateTaskBook(targetList);
+        assertCommandBehavior("edit 1 period: 2015 10 10 10 10:10 to 2015 11 11 11 11:11",
```

Comment:
Why is this supposed to be a successful command? Natty can't handle this date token.